src/Lib.hs

```diff
@@ -23,31 +23,36 @@ import Base.Interval as I
     , intervalToDistance
     )
 import Scale (Scale(..), BaseMode(..), baseModeIntervals)
-import Data.Set(Set(..))
+
+import Control.Monad (foldM)
+import Data.Set (Set(..))
 import qualified Data.Set as S
 import Data.Map.Strict (Map, insert, fromList, toList, (!), delete, (!?))
 import Data.Maybe (fromJust)
 
 
-chordToNotes :: Chord -> [Root]
+chordToNotes :: Chord -> Maybe [Root]
 chordToNotes chord =
-  flip jumpIntervalFromNote (getChordRoot chord) <$> S.toList (chordToIntervals chord)
+  do chordInts <- chordToIntervals chord
+     return $ flip jumpIntervalFromNote (getChordRoot chord) <$> S.toList chordInts
 
 
-chordToIntervals :: Chord -> Set Interval
+chordToIntervals :: Chord -> Maybe (Set Interval)
 chordToIntervals chord =
-  let
-    baseScale = highestNaturalToIntervals (getHighestNatural chord) $ qualityToIntervals $ C.getQuality chord
-    intervals = susIntervals (extendIntervals baseScale $ getExtensions chord) $ getSus chord
-  in
-    foldr S.insert S.empty intervals
+  do qualInts <- qualityToIntervals $ C.getQuality chord
+     baseScale <- highestNaturalToIntervals (getHighestNatural chord) qualInts
+     extendedScale <- extendIntervals baseScale $ getExtensions chord
+     intervals <- susIntervals extendedScale $ getSus chord
+     return $ foldr S.insert S.empty intervals
```

Comment:
I think this best illustrates the sort of things that changed by making the `Interval` constructor able to fail; if you can understand how this function works, the others should fall out similarly. Feel free to leave more comments here if something isn't clear.

I would encourage you to open this up in VSCode and check out the types of everything; that alone can be a great way to see what's going on with more complex monadic code.